samples/transactional-session/aspnetcore-webapi/sample.md

```diff
@@ -1,50 +1,63 @@
+component: TransactionalSession
+reviewed: 2022-08-12
 related:
-- nservicebus/hosting
+  - TODO
 ---
 
 include: webhost-warning
 
-This sample shows how to send messages to an NServiceBus endpoint from an ASP.NET Core WebAPI application.
+This sample shows how to send messages and modify data in a database atomically using the `NServiceBus.TransactionalSession` package. The operations are triggered by an incoming HTTP request to ASP.NET which will manage the `ITransactionalSession` lifetime.
+
+## Prerequisites
+
+- Visual Studio 2019 is required to run this sample.
+- LocalDB support is required to run the sample. A custom connection string needs to be configured otherwise.
 
 ## Running the solution
 
 When the solution is run, a new browser window/tab opens, as well as a console application. The browser will navigate to `http://localhost:58118/`.
 
-An async [WebAPI](https://dotnet.microsoft.com/apps/aspnet/apis) controller handles the request. It creates an NServiceBus message and sends it to the endpoint running in the console application. The message has been processed successfully when the console application prints "Message received at endpoint".
+An async [WebAPI](https://dotnet.microsoft.com/apps/aspnet/apis) controller handles the request. It stores a new document using Entity Framework and sends an NServiceBus message to the endpoint running in the console application. 
 
-## Prerequisites
+The message will be processed by the NServiceBus message handler and print `"Message received at endpoint"` to the console. The message handler also updates the previously created entity.
 
-- Visual Studio 2019 is required to run this sample.
+## Configuration
 
-### Initialize the WebAPI endpoint
+The host is configures NServiceBus with the `UseNServiceBus` extension method:
 
-Open `Program.cs` and look at the `UseNServiceBus` method.
+snippet: txsession-nsb-configuration
 
-A [Send-Only endpoint](/nservicebus/hosting/#self-hosting-send-only-hosting) named `Samples.ASPNETCore.Sender` is configured in the WebAPI application by creating a new `EndpointConfiguration`.
+The configuration enables the transactional session functionalty using `endpointConfiguration.EnableTransactionalSession()`. Note that the transactional session feature also requires the outbox to be enabled.
 
-snippet: EndpointConfiguration
+ASP.NET Core is configured using the `ConfigureWebHostDefaults` extension method. A custom middleware is registered to manage the `ITransactionalSession` lifetime:
 
-Routing is configured to send every message from the assembly containing `MyMessage` to the `Samples.ASPNETCore.Endpoint` endpoint.
+snippet: txsession-web-configuration
 
-### Injection into the Controller
+Entity Framework support is configured by this registration for the `DbContext`:
 
-The message session is injected into `SendMessageController` via constructor injection.
+snippet: txsession-ef-configuration
```

Comment:
This snippet is no longer needed. I think.

Reply:
Why not?

Reply:
My bad sorry.